org.eclipse.lsp4j.debug/src/main/java/org/eclipse/lsp4j/debug/DebugProtocol.java

```diff
@@ -0,0 +1,11631 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Kichwa Coders Ltd. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+
+package org.eclipse.lsp4j.debug;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * <p>
+ * Declaration of parameters, response bodies, and event bodies.
+ * </p>
+ * <p>
+ * Auto-generated from debugProtocol.json schema version 1.24.0. Do not edit
+ * manually.
+ * </p>
+ */
+public class DebugProtocol {
+	/**
+	 * <p>
+	 * Version of debugProtocol.json this class was derived from.
+	 * </p>
+	 */
+	public static final String SCHEMA_VERSION = "1.24.0";
+
+	/**
+	 * <p>
+	 * Event message for 'stopped' event type.
+	 * </p>
+	 * <p>
+	 * The event indicates that the execution of the debuggee has stopped due to
+	 * some condition.
+	 * </p>
+	 * <p>
+	 * This can be caused by a break point previously set, a stepping action has
+	 * completed, by executing a debugger statement etc.
+	 * </p>
+	 */
+	public static class StoppedEventArguments {
```

Comment:
The class structure should not be nested, but flat, in order to be consistent with the LSP implementation.

Reply:
Probably it would be better to generate a `DebugProtocol.xtend` with `@JsonRpcData` tags and that gets us even closer. It would be a 2 stage generation process (first the typescript one then the o.e.lsp4j.generator one), but then the `DebugProtocol.xtend` would be much readable (and shorter). WDYT?

Reply:
Two-stage generation sounds a bit scary at first. But a nice side effect is that all the derived members (getters, setters, equals, hashCode) are consistent in both protocols. And I like the idea to have a more readable version of the Java data structures.